### DISCUSSION TITLE
Ease of Use

### DISCUSSION THREAD

**Discussion by ac-arcana** [blackears/cyclopsLevelBuilder#190] [General]
Hi all,

I love Cyclops, and I think it is amazing to use so far. I feel like it is an adequate replacement for Trenchbroom as the best level editor for Godot. The editing being straight in the actual scene is a HUGE plus to me.

However, I still have some frustrations when using it. I was wondering if anyone has any tips to improve the ease of use and experience with Cyclops? I have some specific questions, but I am also interested in tips in general from the Cyclops Community.

Some points of pain for me that I wish I could fix:

- The default godot gizmo is in my way all the time. Is there a way to toggle it?
- I find when I move the viewport with the keyboard I am hitting shortcuts in Cyclops constantly changing my tools. Is there a way to see and change the shortcuts for Cyclops?

**Comment by blackears** [comment]
I have been working on something to let you manually set the hot keys.  It's in a separate branch right now, but I hope to merge it soon.

There is currently no way to completely disable the default gizmo, but some users find setting the tool to Show List of Selectable Nodes minimizes the interference.
![image](https://github.com/user-attachments/assets/b1f9bc4c-5322-4835-84af-b4876b51c59d)

Most users seem to be navigating with right-click fly, so you might want to try that if you're not already.  Godot editor currently does not let me manually position the camera.

**Reply by ac-arcana** [reply]
I appreciate the tips! I will try using the "show list of selectable nodes" button.

I tend to move around with wasd and qe when editing, so that's why I'm constantly changing tools. Being able to set my own hotkey would definitely help. Right click fly seems like it might be okay as a work around, except I don't think I can go up/down without qe

**Comment by stefan-s29** [comment]
I am going to use this thread as my suggestions also aim at UX aspects.

1. The arrows of the gizmo should light up when you hover over them like the arrows of the Godot transform gizmo do, so you can be sure which dimension you are currently manipulating.
2. It seems that you can also click in between the gizmo elements and it will default to translating the object on the x/z plane. In that case, the square icon belonging to that plane should light up to make it clear.
3. The highlight of the square icons does not work when you use `Show List of Selectable Nodes` to hide the Godot transform gizmo as suggested by you above.
4. I feel that it should also be possible to scale a Cyclops block by using the plane gizmos the way it works in Godot itself (except, of course, that it should only scale block-wise). That would make it possible to extend a block in two dimensions at once without skewing it, and you could resize a block from a top-down view. This would obviously require a new "scale" tool in the toolbar, so maybe it's a subject for a separate discussion.
5. Pressing `F` centers the gizmo of a Cyclops block, which is not necessarily at the center of the object. It would be great if the function could be manipulated to take into account the dimensions of the Cyclops object, or if there was a new hotkey to replace the Godot hotkey for Cyclops.
6. The highlight on the selection type icons (vertex / edge / face) are very subtle and they are usually rather small, which makes it hard to keep track of what mode you are currently working with. A clear indicator in the view pane would be very useful.
7. The highlighting of the selected face is too subtle in my opinion, no matter if environmental light is turned on or off in the scene.
8. As of now, you cannot switch from one Cyclops block to another by clicking it if the vertex / edge / face selection mode is active. It would be great if you could click a face on another block to switch to this block and select that face.
9. I might be wrong, but it seems that you can only show the Cyclops toolbar by selecting an existing block. It should be possible to create a new block using drag&drop even if no Cyclops block is selected.

Sorry, I know this is a lot. 🙂 Are any of these already on your list or being implemented? I think that they would make Cyclops feel a lot more polished.

I might also help with the implementation if I find some time.

**Reply by blackears** [reply]
Those are all good ideas.  Some won't work in the Godot viewport and will have to wait until I create a new editor window for drawing blocks.  Everything is going to have to wait until I have more time to work on this.  Sorry it's taking so long.

**Reply by stefan-s29** [reply]
No worries and thanks for taking the time to read my suggestions!

A dedicated editor window sounds great.

Like I said, I am motivated to help, but just as for you, time is a limiting factor. You know best what is feasible and if there are any features with limited scope that could be delegated. I will follow the development in any case.

**Reply by blackears** [reply]
Most of this should be fairly easy to add.  I think 8 is possible right now by hovering the mouse over the other block you want  and pressing Alt-Q  It might be best to put these suggestions into an issue instead of a discussion so that it will be easier to find later.

**Reply by stefan-s29** [reply]
Sure. What would you suggest? An issue for each individual bullet point or should we group them as far as possible?

**Reply by blackears** [reply]
Individual items would be best.  That way I could address them individually.

**Reply by stefan-s29** [reply]
I just created the issues.

I would have added labels like `enhancement` or `bug`, but it seems that only you can do that.

**Comment by blackears** [comment]
That's okay.  I do want to get to them soon, but I've had a lot  of trouble to find the time to work on this in the last few months.
